src/config.rs

```diff
@@ -116,12 +118,22 @@ struct RawDisplayConfig {
     pub use_pager: bool,
 }
 
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
+struct RawUpdatesConfig {
+    #[serde(default)]
+    pub auto_update: bool,
+    #[serde(default)]
+    pub auto_update_interval_hours: u64,
```

Comment:
With `#[serde(default)]` this will default to 0, causing every invocation to auto update if `auto_update` is set to true. We should probably set an explicit serde default instead. (The default set in `RawConfig::new` only applies if no config file exists, that could be removed if we have an explicit default.)

I think `#[serde(default = ...)]` should work, but haven't tested it.